### PR TITLE
feat: dump encrypted user login to session for troubleshooting failed authentication

### PR DESCRIPTION
**Pull request by meotch**
# Summary of Changes

Dumps encrypted user login to session for troubleshooting failed authentication via internal logs

Fixes https://mxcom.atlassian.net/browse/MC-826

## Public API Additions/Changes

No changes to API

## Downstream Consumer Impact

No impact to downstream consumer

# How Has This Been Tested?

Added underlying unit test

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: mdx-web/src/main/java/com/mx/path/model/mdx/web/controller/AuthenticationController.java [hunk 1]
```diff
@@ -194,8 +200,19 @@ public final ResponseEntity<ForgotUsername> forgotUsername(@RequestBody ForgotUs
...
   }
 
+  // Package private
+
+  final void dumpEncryptedLoginHashToSession(String login) {
+    if (Strings.isNotBlank(login)) {
+      String sha256hex = Hashing.sha256()
+          .hashString(login, StandardCharsets.UTF_8)
+          .toString();
+      Session.current().sput(Scope.Session, "login", sha256hex);
```
Comment (RyanJones08) [inline mdx-web/src/main/java/com/mx/path/model/mdx/web/controller/AuthenticationController.java:210]: Do we need to name this field different so when others pull it from session they know it is hashed

Comment (gerfboy) [reply mdx-web/src/main/java/com/mx/path/model/mdx/web/controller/AuthenticationController.java:210]: ooh, good idea. Maybe 'loginHash', or something better. Other than that, I like this idea.

Comment (meotch) [reply mdx-web/src/main/java/com/mx/path/model/mdx/web/controller/AuthenticationController.java:210]: very good call out! updating

Comment (RyanJones08) [reply mdx-web/src/main/java/com/mx/path/model/mdx/web/controller/AuthenticationController.java]: Looks good

**Review by RyanJones08 — APPROVED** [review]

**Review by gerfboy — APPROVED** [review]

**Pull request merged**